libraries/AP_BattMonitor/AP_BattMonitor_Logging.cpp

```diff
@@ -0,0 +1,70 @@
+#include "AP_BattMonitor_Backend.h"
+#include <AP_Logger/AP_Logger.h>
+
+extern const AP_HAL::HAL& hal;
+
+// Write BAT data packet(s)
+void AP_BattMonitor_Backend::Log_Write_BAT(const uint8_t instance, const uint64_t time_us) const
+{
+    bool has_temp = (AP_HAL::millis() - _state.temperature_time) <= AP_BATT_MONITOR_TIMEOUT;
+    bool has_current1 = has_current();
+
+    const struct log_BAT pkt{
+        LOG_PACKET_HEADER_INIT(LOG_BAT_MSG),
+        time_us             : time_us,
+        instance            : instance,
+        voltage             : _state.voltage,
+        voltage_resting     : _state.voltage_resting_estimate,
+        current_amps        : has_current1 ? _state.current_amps : AP::logger().quiet_nanf(),
+        current_total       : has_current1 ? _state.consumed_mah : AP::logger().quiet_nanf(),
+        consumed_wh         : has_current1 ? _state.consumed_wh : AP::logger().quiet_nanf(),
+        temperature         : (int16_t)(has_temp ? (_state.temperature * 100) : 0),
```

Comment:
```suggestion
        temperature         :  get_temerature(temp, instance) ? temp * 100 : 0,
```
Use the helper which manages the time for us, and all we need is a float storage for the temp declared earlier.

Reply:
added has_temp to backend